### PR TITLE
Adds unit tests

### PR DESCRIPTION
**Pull request by ashfurrow**
Hey there! This is a work-in-progress, but as part of the [CocoaPods Test Jam](https://cocoapods.org/testjam), I've added some unit tests to AFNetworkActivityLogger. 

- [x] Add test harness project.
- [x] Abstract out NSLog invocations.
- [x] Replace `equal()` matcher with `matches()` regex matcher (once https://github.com/specta/expecta/pull/144 is merged).
- [x] Finish remaining pending tests.
- [ ] ~~~Test NSLog invocation.~~~
- [x] Add Travis CI integration.
- [ ] ~~~Add code coverage metrics.~~~

### CODE REVIEW

**Comment by ashfurrow** [comment]
So, testing NSLog invocation is actually really hard so I'm leaving it as pending. I'll get Travis + metrics integrated tomorrow.

**Comment by ashfurrow** [comment]
Also going to leave code coverage to someone else – want to get this in a mergeable state ASAP. Here's a [good guide](http://list.her.sh/ios-test-coverage-with-coveralls/) for anyone coming along later.

**Comment by ashfurrow** [comment]
So this is ready for review. You'll need to enable Travis support on your repository from Travis' UI to get the tests running.

**Comment by ashfurrow** [comment]
Help getting Travis to run would be :+1: too. Apparently it's still awful.

File: Tests/AFNetworkActivityLogger.xcworkspace/xcshareddata/AFNetworkActivityLogger.xccheckout [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+	<key>IDESourceControlProjectFavoriteDictionaryKey</key>
+	<false/>
+	<key>IDESourceControlProjectIdentifier</key>
+	<string>628B9B99-905E-44E4-9CA9-B03C47194B1D</string>
+	<key>IDESourceControlProjectName</key>
+	<string>AFNetworkActivityLogger</string>
+	<key>IDESourceControlProjectOriginsDictionary</key>
+	<dict>
+		<key>E3CA97A4D0A57AEDFA77BE925E24F5813E165E2C</key>
+		<string>https://github.com/ashfurrow/AFNetworkActivityLogger.git</string>
```
Comment (kylef) [inline Tests/AFNetworkActivityLogger.xcworkspace/xcshareddata/AFNetworkActivityLogger.xccheckout]: This file should not be committed, it's local checkout information.

Comment (ashfurrow) [inline Tests/AFNetworkActivityLogger.xcworkspace/xcshareddata/AFNetworkActivityLogger.xccheckout]: whoops

Comment (ashfurrow) [inline Tests/AFNetworkActivityLogger.xcworkspace/xcshareddata/AFNetworkActivityLogger.xccheckout]: Cool.

**Comment by ashfurrow** [comment]
@kylef Any update on this?

**Comment by kcharwood** [comment]
I just turned travis on for this repo. Would you want to retarget this pr to the 3.0.0 branch? Prepping that in conjunction with AFNetworking 3.0

**Comment by ashfurrow** [comment]
OK, will do.

**Pull request closed without merging**